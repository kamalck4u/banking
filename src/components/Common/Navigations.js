import React from 'react';
import { useHistory } from 'react-router-dom';

const Navigations = ({ home, back }) => {
	const history = useHistory();
	return (
		<div style={{ display: 'inline' }}>
			{
				home && <i className="fa fa-home fa-2x nav-icon" aria-hidden="true" onClick={() => history.push(home)} />
			}
			{
				back && <i className="fa fa-chevron-left fa-2x nav-icon" aria-hidden="true" onClick={() => history.goBack()} />
			}

		</div>
	);
};

export default Navigations;

