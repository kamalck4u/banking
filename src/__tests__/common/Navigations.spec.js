import React from 'react';
import renderer from 'react-test-renderer';
import Navigations from '../../components/Common/Navigations';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

describe('Navigations', () => {
	it('snapshot renders navigations', () => {
		const HeaderPanelComponent = renderer.create(<Navigations />);
		expect(HeaderPanelComponent.toJSON()).toMatchSnapshot();
	});
});


describe('Navigations', () => {
	let wrapper;
	it('should show home button', () => {
		wrapper = shallow(<Navigations home='/accounts' />);
		expect(wrapper.find('i.fa-home')).toHaveLength(1);
	});
	it('should show back button', () => {
		wrapper = shallow(<Navigations back='/accounts/1/transactions' />);
		expect(wrapper.find('i.fa-chevron-left')).toHaveLength(1);
	});

	it('should show both home and back button', () => {
		wrapper = shallow(<Navigations home='/accounts' back='/accounts/1/transactions' />);
		expect(wrapper.find('i.fa-home')).toHaveLength(1);
		expect(wrapper.find('i.fa-chevron-left')).toHaveLength(1);
	});
});
