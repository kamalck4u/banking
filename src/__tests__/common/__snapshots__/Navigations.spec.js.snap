// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigations snapshot renders navigations 1`] = `
<div
  style={
    Object {
      "display": "inline",
    }
  }
/>
`;
